Algorithm.Python/AddOptionContractFromUniverseRegressionAlgorithm.py

```diff
@@ -14,65 +14,65 @@
+### Later we call 'QCAlgorithm.remove_option_contract' and expect both option and underlying to be removed.
 ### </summary>
 class AddOptionContractFromUniverseRegressionAlgorithm(QCAlgorithm):
-    def Initialize(self):
+    def initialize(self):
         '''Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.'''
 
-        self.SetStartDate(2014, 6, 5)
-        self.SetEndDate(2014, 6, 9)
+        self.set_start_date(2014, 6, 5)
+        self.set_end_date(2014, 6, 9)
 
         self._expiration = datetime(2014, 6, 21)
-        self._securityChanges = None
+        self._security_changes = None
         self._option = None
         self._traded = False
 
-        self._twx = Symbol.Create("TWX", SecurityType.Equity, Market.USA)
-        self._aapl = Symbol.Create("AAPL", SecurityType.Equity, Market.USA)
-        self.UniverseSettings.Resolution = Resolution.Minute
-        self.UniverseSettings.DataNormalizationMode = DataNormalizationMode.Raw
+        self._twx = Symbol.create("TWX", SecurityType.EQUITY, Market.USA)
+        self._aapl = Symbol.create("AAPL", SecurityType.EQUITY, Market.USA)
+        self.universe_settings.resolution = Resolution.MINUTE
+        self.universe_settings.data_normalization_mode = DataNormalizationMode.RAW
 
-        self.AddUniverse(self.Selector, self.Selector)
+        self.add_universe(self.selector, self.selector)
 
-    def Selector(self, fundamental):
-        if self.Time <= datetime(2014, 6, 5):
+    def selector(self, fundamental):
+        if self.time <= datetime(2014, 6, 5):
             return  [ self._twx ]
         return [ self._aapl ]
 
-    def OnData(self, data):
+    def on_data(self, data):
         '''OnData event is the primary entry point for your algorithm. Each new data point will be pumped in here.
 
         Arguments:
             data: Slice object keyed by symbol containing the stock data
         '''
-        if self._option != None and self.Securities[self._option].Price != 0 and not self._traded:
+        if self._option != None and self.securities[self._option].price != 0 and not self._traded:
             self._traded = True
-            self.Buy(self._option, 1)
+            self.buy(self._option, 1)
 
-        if self.Time == datetime(2014, 6, 6, 14, 0, 0):
+        if self.time == datetime(2014, 6, 6, 14, 0, 0):
             # liquidate & remove the option
-            self.RemoveOptionContract(self._option)
+            self.remove_option_contract(self._option)
 
-    def OnSecuritiesChanged(self, changes):
+    def on_securities_changed(self, changes):
         # keep track of all removed and added securities
-        if self._securityChanges == None:
-            self._securityChanges = changes
+        if self._security_changes == None:
+            self._security_changes = changes
         else:
-            self._securityChanges.op_Addition(self._securityChanges, changes)
```

Comment:
Not related to PEP8 but if SecurityChanges defines the `+` operator, it should be accessible in Python without explicitly calling `op_Addition`